src/sentry/apidocs/hooks.py

```diff
@@ -222,6 +225,12 @@ def custom_postprocessing_hook(result: Any, generator: Any, **kwargs: Any) -> An
                     else:
                         schema = content["application/json"]["schema"]
 
+                    # If the schema is a reference, we need to resolve it
+                    if len(schema) == 1 and "$ref" in schema:
```

Comment:
would be good to split this out and add a test

Reply:
I moved the function into a helper. Regarding testing, I tried using the `generate_schema` function we copied over from drf-spec with the test `ExampleEndpoint`, but for some reason the `requestBody` field isn't building in the schema. It works in the drf-spec tests tho so I'm not sure what we're missing 😐

Reply:
sorry, meant we should split this into a separate PR. but helper function is good too.

Reply:
whoops ok, new PR is here: https://github.com/getsentry/sentry/pull/57639